pkg/systemd/entry.go

```diff
@@ -0,0 +1,99 @@
+package systemd
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// Entry holds a systemd log entry.
+// https://www.freedesktop.org/software/systemd/man/systemd.journal-fields.html
+type Entry struct {
+	// Hostname is the name of the originating host.
+	Hostname string
+
+	// InvocationID is a randomized, unique 128-bit ID identifying each
+	// runtime cycle of the unit.
+	InvocationID string
+
+	// MessageID is a 128bit message ID for recognizing certain message
+	// types.
+	MessageID string
+
+	// Message it the human-readable message string for this entry. This
+	// is supposed to be the primary text shown to the user. It is
+	// usually not translated (but might be in some cases), and is
+	// not supposed to be parsed for metadata.
+	Message string
+
+	// PID is the process ID of the process the journal entry originates
+	// from.
+	PID int
+
+	// Restarts is a counter of unit restarts, as long as the unit isn't
+	// fully stopped, i.e. as long as it remains up or remains in
+	// auto-start states.
+	Restarts int
+
+	// SourceRealtimeTimestamp is the earliest trusted timestamp of the
+	// message, if any is known that is different from the reception time
+	// of the journal.
+	SourceRealtimeTimestamp time.Time
+
+	// SyslogIdentifier is the identifier string (i.e.  "tag") as
+	// specified in the original datagram.
+	SyslogIdentifier string
+
+	// Unit FIXME: spec?
```

Comment:
@cgwalters : do you know of any docs for `UNIT=`?  I didn't see anything [in the docs][1], but it's definitely part of log entries we are pulling down off the cluster.

[1]: https://github.com/systemd/systemd/blob/v246/man/systemd.journal-fields.xml

Reply:
I agree it should probably be mentioned there but...that's because `UNIT` is custom to just some types of log messages from systemd; it won't appear in anything close to all messages.

Anything using structured journaling can add whatever fields they want at each logging site; systemd adds `UNIT` for those log messages that are relevant.  Other projects can add their own fields.

So think of that doc more like "core fields", and other places in systemd just happen to use structured logging.  That said I am quite sure the systemd crew would think of the `USER` field as an ABI, it's not going to go away.

Perhaps worth filing an issue upstream.

Reply:
Filed upstream: systemd/systemd#17538.